### PR TITLE
Fix for #9 of Part of the site doesn't use translated strings #4522

### PR DESCRIPTION
**Pull request by NathanMaton**
Fix for #9 of Part of the site doesn't use translated strings #4522

### CODE REVIEW

**Comment by paglias** [comment]
@NathanMaton thanks for the PR!

The build is broken which means some tests are failing after your changes, also why did you add a new package that is not used anywhere and those 2 files to gitignore? 

As for the code itself, you should move the env.t call from the jade file directly to the JavaScript one so that the texts always show he new strings, not only in the template

File: common/script/content/mystery-sets.js [hunk 1]
```diff
@@ -4,122 +4,122 @@ let mysterySets = {
   201402: {
     start: '2014-02-22',
     end: '2014-02-28',
-    text: 'Winged Messenger Set',
+    text: 'wingedMessengerSet',
```
Comment (crookedneighbor) [inline common/script/content/mystery-sets.js]: This the linting error that caused the build to fail. A trailing blank space.

File: common/locales/en/subscriber.json [hunk 2]
```diff
@@ -63,6 +63,30 @@
     "timeTravelersPopover": "We see you have a Mystic Hourglass, so we will happily travel back in time for you! Please choose the pet, mount, or Mystery Item Set you would like. You can see a list of the past item sets <a href='http://habitica.wikia.com/wiki/Mystery_Item' target='_blank'>here</a>! If those don't satisfy you, perhaps you'd be interested in one of our fashionably futuristic Steampunk Item Sets?",
     "timeTravelersAlreadyOwned": "Congratulations! You already own everything the Time Travelers currently offer. Thanks for supporting the site!",
     "mysticHourglassPopover": "A Mystic Hourglass allows you to purchase certain limited-time items, such as monthly Mystery Item Sets and awards from world bosses, from the past!",
+    "wingedMessengerSet":"Winged Messenger Set",
```
Comment (crookedneighbor) [inline common/locales/en/subscriber.json]: I think these should be namespaced with the prefix "mysterySet" so you'd have:

``` json
"mysterySetWingedMessenger": "Winged Messenger Set",
"mysterySetForestWalker": "Forest Walker Set",
// ...
```

You could even do:

``` json
"mysterySet201402": "Winged Messenger Set",
"mysterySet201403": "Forest Walker Set",
// etc
```

And then you could remove the text attribute from the objects, and just loop over the mystery sets object (line 130-132) to set each text attribute to be:

```
// top of the file
let t = require('./translation.js');

// line 130
each(mysterySets, (value, key) => {
  value.key = key;
  value.text = t(`mysterySet${key}`);
});
```

File: website/views/options/inventory/time-travelers.jade [hunk 3]
```diff
@@ -29,7 +29,7 @@
                 popover-placement='right', popover-append-to-body='true',
                 ng-click='clickTimeTravelItem("#{type}",item)')
       li.customize-menu.inventory-gear
-        menu.pets-menu(label='{{::set.text}}', ng-repeat='set in Content.timeTravelerStore(user.items.gear.owned)')
+        menu.pets-menu(label=env.t('{{::set.text}}'), ng-repeat='set in Content.timeTravelerStore(user.items.gear.owned)')
```
Comment (crookedneighbor) [inline website/views/options/inventory/time-travelers.jade]: If you set the translation function in the content object, then all you need to do is: `label='{{::set.text()}}'`

File: common/locales/en/subscriber.json (hunk 2 above)
Comment (Alys) [inline common/locales/en/subscriber.json]: +++1 for the "mysterySet201402" format. Much preferred when working out problems related to people not getting their items.

Comment (Alys) [inline common/locales/en/subscriber.json]: Also excellent for sorting, which could be useful.

**Comment by NathanMaton** [comment]
Thanks for the comments (this is my first PR so that is why a few things may be off).

The package seemed to come from another update on the /develop branch when I pulled it down. Should I just exclude that? For the .gitignore stuff I can exclude that.

For the code what's the right way to call the env.t command in the javascript file? I saw elsewhere in the app we use window.env.t('') but when I tried that on the mystery-sets.js file it didn't run locally.

File: common/script/content/mystery-sets.js [hunk 4]
```diff
@@ -129,6 +132,7 @@ let mysterySets = {
 
 each(mysterySets, (value, key) => {
   value.key = key;
+  value.text = t('mysterySet${key}'');
```
Comment (crookedneighbor) [inline common/script/content/mystery-sets.js]: These should be backticks (`), not single quotes ('). See http://babeljs.io/docs/learn-es2015/#template-strings

You also have an extra single quote in there.

File: package.json [hunk 5]
```diff
@@ -43,6 +43,7 @@
     "in-app-purchase": "^0.2.0",
     "jade": "~1.11.0",
     "js2xmlparser": "~0.1.2",
+    "latest-version": "^2.0.0",
```
Comment (crookedneighbor) [inline package.json]: This should be removed.

**Comment by crookedneighbor** [comment]
Correct the syntax error, remove the extraneous package from the package.json and removed the commented out code, and this should be good to go.

File: common/script/content/mystery-sets.js (hunk 4 above)
Comment (NathanMaton) [inline common/script/content/mystery-sets.js]: Thanks for the note!

**Comment by NathanMaton** [comment]
Put in requested edits. Build still failing locally for me but out of time to debug tonight.

**Comment by crookedneighbor** [comment]
Merged in with https://github.com/HabitRPG/habitrpg/commit/35a921bf43a1d33b0606e622d5ab8fd809d834b1

This will count towards a contributor tier! What's your User ID? You can find it in [Settings > API](https://habitrpg.com/#/options/settings/api). Be sure to only copy the User ID and _not_ your API Token.

**Comment by NathanMaton** [comment]
Thanks Blade. UiD is 2ab55d89-a3e5-4565-ba35-b6c9e145a60a

Definitely hope to contribute more.

**Comment by crookedneighbor** [comment]
Tier awarded! You can read more about [contributor tiers and rewards here](http://habitrpg.wikia.com/wiki/Contributor_Rewards).

Note that as your tier increases, it takes progressively more work to achieve the next one, so for the higher tiers, larger PRs or a few PRs are required. There's some information about that at Contributor Rewards, but feel free to ask here if you have questions!

**Comment by NathanMaton** [comment]
Thanks!

**Pull request closed without merging**